torch/distributed/pipelining/schedules.py

```diff
@@ -1041,6 +1041,77 @@ def __init__(
                 "Simply stop passing it, and everything should still work fine."
             )
 
+    def seeded_module_init(
+        self,
+        stage_initializer: Callable[[torch.nn.Module], None],
```

Comment:
should 'stage_initializer' be an optional argument?  I could check stage.submod for the presence of an 'init_weights' method, and if present, use that as the default.  Otherwise, if 'stage_initializer' is not provided, I would raise an error.